nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/AbstractPutHBase.java

```diff
@@ -131,6 +133,59 @@ public void onScheduled(final ProcessContext context) {
         clientService = context.getProperty(HBASE_CLIENT_SERVICE).asControllerService(HBaseClientService.class);
     }
 
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (propertyDescriptorName.startsWith("visibility.")) {
+            String[] parts = propertyDescriptorName.split("\\.");
+            String displayName;
+            String description;
+
+            if (parts.length == 2) {
+                displayName = String.format("Column Family %s Default Visibility", parts[1]);
+                description = String.format("Default visibility setting for %s", parts[1]);
+            } else if (parts.length == 3) {
+                displayName = String.format("Column Qualifier %s.%s Default Visibility", parts[1], parts[2]);
+                description = String.format("Default visibility setting for %s.%s", parts[1], parts[2]);
+            } else {
+                return null;
+            }
+
+            return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .displayName(displayName)
+                .description(description)
+                .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+                .dynamic(true)
+                .build();
+        }
+
+        return null;
+    }
+
+    protected String pickVisibilityString(String columnFamily, String columnQualifier, FlowFile flowFile, ProcessContext context) {
+        final String lookupKey = String.format("visibility.%s.%s", columnFamily, columnQualifier);
+        final String fromAttribute = flowFile.getAttribute(lookupKey);
+        if (fromAttribute != null) {
+            return fromAttribute;
+        } else {
+            PropertyValue descriptor = context.getProperty(lookupKey);
+            if (descriptor == null) {
```

Comment:
This condition does not work as expected. Because even if there is only 'visibility.family' defined, `context.getProperty(visibility.family.qualifier)` returns a non-null object which having its value null. Please change this to `(descriptor == null || !descriptor.isSet())`. Also please add some unit tests.